cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java

```diff
@@ -99,4 +105,20 @@ public void testStreamConversion() throws Exception {
     Assert.assertEquals(minute, results.getInt(5));
     Assert.assertFalse(results.next());
   }
+
+  private <T> T testWithRetry(Callable<T> callable, int timesToRetry,
```

Comment:
Also `testWithRetry` is meant to allow multiple assertions inside of the `callable`, so it doesn't really fit the use case as it's not really waiting for a single value.

Perhaps we could add a way to specify exception types to ignore to `Tasks.waitFor`, but I don't think that's necessary for this PR.